### ISSUE TITLE
Game page does not open if Heroic can't access the GOG website

### ISSUE THREAD

**Issue by arielj** [Heroic-Games-Launcher/HeroicGamesLauncher#1613]
### Describe the bug

I'm not sure if my description is correct, but currently the GOG site is down and when I click on a game to show the game page or the game settings, Heroic stays loading and loading and nothing happens.

I imagine something similar happens if the internet connection is not working since in practice it would mean Heroic can connect to the GOG site.

### Add logs

First thing I see in the logs (without clicking the game card):

```
(21:51:18) INFO:    [Gog]:             Refreshing access_token
(21:54:18) ERROR:   [Gog]:             Error with refreshing token, reauth required
(node:1315542) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'access_token')
    at GOGUser.getUserDetails (/home/ariel/dev/oss/HeroicGamesLauncher/public/gog/user.js:45:47)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
(node:1315542) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(21:54:18) ERROR:   [Gog]:             Error with refreshing token, reauth required
```

Then this appears (I think it's trying to check for updates of the game, but it doesn't have the game info?):

```
Error occurred in handler for 'checkGameUpdates': TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:371:5)
    at validateString (node:internal/validators:120:11)
    at Url.parse (node:url:169:3)
    at Object.urlParse [as parse] (node:url:156:13)
    at dispatchHttpRequest (/home/ariel/dev/oss/HeroicGamesLauncher/node_modules/axios/lib/adapters/http.js:124:22)
    at new Promise (<anonymous>)
    at httpAdapter (/home/ariel/dev/oss/HeroicGamesLauncher/node_modules/axios/lib/adapters/http.js:48:10)
    at dispatchRequest (/home/ariel/dev/oss/HeroicGamesLauncher/node_modules/axios/lib/core/dispatchRequest.js:58:10)
    at Axios.request (/home/ariel/dev/oss/HeroicGamesLauncher/node_modules/axios/lib/core/Axios.js:108:15)
    at Axios.<computed> [as get] (/home/ariel/dev/oss/HeroicGamesLauncher/node_modules/axios/lib/core/Axios.js:129:17) {
  code: 'ERR_INVALID_ARG_TYPE'
}
(21:54:21) ERROR:   [Frontend]:        Error: Error invoking remote method 'checkGameUpdates': TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received undefined
```

Then when I click the card:

In the backend I see:
```
(21:56:45) ERROR:   [Backend]:         TypeError: Cannot read properties of null (reading '_embedded')
```

and in the dev tools console I see:
```
error: TypeError: Cannot destructure property 'install' of 'newInfo' as it is undefined. at updateConfig (http://localhost:3000/static/js/bundle.js:8416:11)
message: "Cannot destructure property 'install' of 'newInfo' as it is undefined."
stack: "TypeError: Cannot destructure property 'install' of 'newInfo' as it is undefined.\n    at updateConfig (http://localhost:3000/static/js/bundle.js:8416:11)"
```

and the loader keeps spinning forever

### Steps to reproduce

1. Have a GOG game installed
2. At the moment of reporting this, the GOG site is down, but I can imagine you can disconnect from the internet or block the access or whatever
3. Open Heroic
4. Click the game card

### Expected behavior

The game card should show fine, you should be able to change settings too (to set the Offline mode for example?).

### System Information

- OS [e. g. "Ubuntu"]: Linux Mint
- Heroic Version [e. g. 2.1.1]: 2.4.0-beta.2

### Additional information

More people reported this on Discord.

**Comment by Knut01** [comment]
Hello.
Same problem with version 2.3.10. However 2.3.9 works on Ubuntu 22.04.

**Comment by MrMinimal** [comment]
Same behaviour when you disable your wifi (on Steam Deck for example). When selecting a game the UI is stuck saying 'loading'.

I'd edit the title to "Can't play GOG games without internet connection".

**Comment by flavioislima** [comment]
Fixed 2.4.0
